src/Elliptic/Systems/Elasticity/BoundaryConditions/Zero.cpp

```diff
@@ -0,0 +1,85 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Elliptic/Systems/Elasticity/BoundaryConditions/Zero.hpp"
+
+#include <cstddef>
+#include <string>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Elliptic/BoundaryConditions/BoundaryCondition.hpp"
+#include "Elliptic/BoundaryConditions/BoundaryConditionType.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/Gsl.hpp"
+
+namespace Elasticity::BoundaryConditions::detail {
+
+template <size_t Dim, elliptic::BoundaryConditionType BoundaryConditionType>
+std::string ZeroImpl<Dim, BoundaryConditionType>::name() noexcept {
+  if constexpr (BoundaryConditionType ==
+                elliptic::BoundaryConditionType::Dirichlet) {
+    return "Fixed";
+  } else {
+    return "Free";
+  }
+}
+
+template <size_t Dim, elliptic::BoundaryConditionType BoundaryConditionType>
+void ZeroImpl<Dim, BoundaryConditionType>::apply(
+    const gsl::not_null<tnsr::I<DataVector, Dim>*> displacement,
+    const gsl::not_null<tnsr::I<DataVector, Dim>*>
+        n_dot_minus_stress) noexcept {
+  if constexpr (BoundaryConditionType ==
+                elliptic::BoundaryConditionType::Dirichlet) {
+    (void)n_dot_minus_stress;
+    for (size_t d = 0; d < Dim; ++d) {
+      displacement->get(d) = 0.;
+    }
+  } else {
```

Comment:
do you want to check that this to have a `else if constexpr(... = Neumann)` and and then a generic else with a static assert in case BoundaryConditionType gets extended?

Reply:
That should already be covered by the static assert in the header (line 43), don't you think?

Reply:
should be as long as if you remember to modify both if an additional choice is added to the list.  Up to you

Reply:
yea should be fine